quickwit/quickwit-control-plane/src/ingest/ingest_controller.rs

```diff
@@ -365,6 +417,147 @@ impl IngestController {
             }
         }
     }
+
+    /// Attempts to increase the number of shards.
+    async fn try_scale_out_shards(
+        &mut self,
```

Comment:
can we have counters about the number of shards per source or do you think this will be too much?

Reply:
A gauge per source seems like a good idea.